src/pages/Browse/index.spec.tsx

```diff
@@ -542,4 +548,157 @@ describe(__filename, () => {
 
     expect(root.find('title')).toHaveText(`Browse ${name}: ${versionString}`);
   });
+
+  describe('preloading', () => {
+    const file1 = 'file1.js';
+    const file2 = 'file2.js';
+    const setUpFilesAndRender = ({
+      beginFetchNextFile = false,
+      buildTree = true,
+      loadNextFile = false,
+      loadVersionFile = true,
+    }: {
+      beginFetchNextFile?: boolean;
+      buildTree?: boolean;
+      loadNextFile?: boolean;
+      loadVersionFile?: boolean;
+    } = {}) => {
+      const extraFileEntries = {
+        [file1]: { ...fakeVersionEntry, path: file1 },
+        [file2]: { ...fakeVersionEntry, path: file2 },
+      };
+      const {
+        _fetchVersionFile,
+        addonId,
+        fakeThunk,
+        store,
+        version,
+      } = setUpVersionFileUpdate({
+        extraFileEntries,
+        initialPath: file1,
+        loadVersionFile,
+      });
+      if (buildTree) {
+        store.dispatch(
+          fileTreeActions.buildTree({
+            version: createInternalVersion(version),
+          }),
+        );
+      }
+      if (beginFetchNextFile) {
+        store.dispatch(
+          versionsActions.beginFetchVersionFile({
+            path: file2,
+            versionId: version.id,
+          }),
+        );
+      }
+      if (loadNextFile) {
+        store.dispatch(
+          versionsActions.loadVersionFile({ path: file2, version }),
+        );
+      }
+      const dispatchSpy = spyOn(store, 'dispatch');
+
+      render({
+        _fetchVersionFile,
+        addonId: String(addonId),
+        store,
+        versionId: String(version.id),
+      });
+
+      return {
+        _fetchVersionFile,
+        addonId,
+        dispatchSpy,
+        fakeThunk,
+        store,
+        version,
+      };
+    };
+
+    it('dispatches fetchVersionFile for the next file', () => {
+      const {
+        _fetchVersionFile,
+        addonId,
+        fakeThunk,
+        version,
+        dispatchSpy,
+      } = setUpFilesAndRender();
+
+      expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+      expect(_fetchVersionFile).toHaveBeenCalledWith({
+        addonId,
+        path: file2,
```

Comment:
The helper is looking good but consider how `file2` is used in this assertion. This means that the test depends on the value of `file2` and, thus, it should control the value. In this patch, the test does not control the value (it relies on a shared default). This makes the test hard to read and it makes the shared helper hard to edit for future enhancements.

Try to make `file2` an input to the helper function so that the test can control it. Example:

```diff
diff --git a/src/pages/Browse/index.spec.tsx b/src/pages/Browse/index.spec.tsx
index 414e1ac..07499f2 100644
--- a/src/pages/Browse/index.spec.tsx
+++ b/src/pages/Browse/index.spec.tsx
@@ -550,16 +550,18 @@ describe(__filename, () => {
   });
 
   describe('preloading', () => {
-    const file1 = 'file1.js';
-    const file2 = 'file2.js';
     const setUpFilesAndRender = ({
       beginFetchNextFile = false,
       buildTree = true,
+      file1 = 'file1.js',
+      file2 = 'file2.js',
       loadNextFile = false,
       loadVersionFile = true,
     }: {
       beginFetchNextFile?: boolean;
       buildTree?: boolean;
+      file1?: string;
+      file2?: string;
       loadNextFile?: boolean;
       loadVersionFile?: boolean;
     } = {}) => {

```

As for `addonId` and `versionId`, the test should control these as inputs, too.